### DISCUSSION TITLE
conceptual question and plots as in figure 4C

### DISCUSSION THREAD

**Discussion by andrea-de-micheli** [jackbibby1/SCPA#20] [General]
Hi Jack, I'm really enjoying your work and it is providing valuable insights for our project.

The way I'm integrating SCPA into my pipeline is the following and I just wanted to run it by you for suggestions or any conceptual errors. I'm not using the MSigDB.

- I have two groups I want to compare (treated vs. untreated)
- Per group, I extract the top 1000 most variable genes (Seurat::FindVariableGenes).
- I run g:profiler (gprofiler2::gost) on the top 1000 most variable genes per group independently.
- I combine the enriched gene sets from both independent g:profiler analysis and I create a custom GMT file as per your tutorial.
- I run SCPA with my custom GMT to identify gene sets that are differentially regulated.

I could run SCPA on the entire GO/KEGG/etc database, but since there are over 30k gene sets, I prefer to focus on those that are also significantly enriched.

Does my approach seems reasonable to you?

I also have a pseudo time timestamp for every cells, so was also wondering if you could provide the code for generating Fig 4C (preprint). For me this representation is the most intuitive for visualising that a pathway is differentially regulated.
Thank you very much!

Best,
Andrea

**Comment by jackbibby1** [comment]
Hi Andrea,

Thanks -- good to hear. 

Because SCPA doesn't consider pathway enrichment as the only factor that should be important when looking at pathway activity, I wouldn't only focus only on enriched gene sets. There's a brief section on the SCPA output [here](https://jackbibby1.github.io/SCPA/articles/interpreting_scpa_output.html#which-output-values-should-you-use) that discusses what the Qval represents, and why non-enriched pathways are important to consider. But in brief, SCPA will detect both enriched pathways, and pathways that show transcriptional changes independent of enrichment, which we know are also important for cellular function (e.g. Figure 4D-J in our paper). More broadly, I think using SCPA on an unfiltered gene set list -- e.g. all the canonical pathways from msigdb -- will give you the best chance of finding the most appropriate pathway signal.

I have a repo with all the scripts to replicate the main figures [here](https://github.com/jackbibby1/scpa_paper), and you can get the specific pseudotime analysis from [this script](https://github.com/jackbibby1/scpa_paper/blob/main/figure_4.R) on lines 82-124. The scripts in this repo are related to the published [Cell Reports version](https://www.cell.com/cell-reports/fulltext/S2211-1247(22)01571-6) and not the preprint, but I think it should mostly be the same.

Just let me know if there's anything else

Jack

**Reply by jackbibby1** [reply]
Apologies -- edited the above text to correct the lines referenced in the script

**Reply by andrea-de-micheli** [reply]
Thank you Jack for your answer. I understand why it makes more sense to run SCPA on unfiltered gene sets.

Best,
Andrea
